### PR TITLE
[INLONG-5040][Sort] Add Apache Doris extract node for Sort

### LINKED ISSUE #5040: [Feature][Sort] Add Apache Doris extract node for Sort
### Description

you can refer to:
https://inlong.apache.org/docs/next/design_and_concept/how_to_extend_data_node_for_sort#extend-a-new-extract-node

### Use case

_No response_

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by duowan1520**
### Prepare a Pull Request
- Fixes #5040 

### Motivation

Add Apache Doris Extract Node for Sort

### Modifications
- Add doris Extract Node
- Add apache doris official doris-flink-connector
- Add doris Extract Node test case

### Verifying this change
- [x] This change added tests and can be verified as follows:
- add the Unit test of DorisExtractLoad DorisExtractNodeTest
- add the Unit test of DorisExtractLoad DorisExtractToDorisLoadTest

### CODE REVIEW

**Review by yunqingmoswu — APPROVED** [review]
LGTM

File: inlong-distribution/src/main/assemblies/sort-connectors.xml [hunk 1]
```diff
@@ -1,159 +1,167 @@
...
+            <directory>../inlong-sort/sort-connectors/doris/target</directory>
+            <outputDirectory>inlong-sort/connectors</outputDirectory>
+            <includes>
+                <include>sort-connector-doris-${project.version}.jar</include>
+            </includes>
+            <fileMode>0644</fileMode>
+        </fileSet>
+        <!-- module's 3td-licenses, notices-->
+        <fileSet>
+            <directory>../licenses/inlong-sort-connectors</directory>
```
Comment (dockerzhang) [inline inlong-distribution/src/main/assemblies/sort-connectors.xml:160]: please keep the same format for this file.

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/DorisExtractNode.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    @JsonProperty("username")
+    @Nonnull
+    private String userName;
+
+    @JsonProperty("password")
+    @Nonnull
+    private String password;
+
+    @JsonProperty("table.identifier")
```
Comment (EMsnap) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/DorisExtractNode.java:59]: tableIdentifier maybe proper

File: inlong-distribution/src/main/assemblies/sort-connectors.xml (hunk 1 above)
Comment (duowan1520) [reply inlong-distribution/src/main/assemblies/sort-connectors.xml:160]: Ok, I fixed it in #INLONG-5041

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/DorisExtractNode.java (hunk 2 above)
Comment (duowan1520) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/DorisExtractNode.java]: ok, i've dealt with it

File: inlong-distribution/src/main/assemblies/sort-connectors.xml [hunk 3]
```diff
@@ -1,159 +1,167 @@
...
-        <fileSet>
-            <directory>../licenses/inlong-sort-connectors</directory>
-            <includes>
-                <include>**/*</include>
-            </includes>
-            <outputDirectory>./</outputDirectory>
-        </fileSet>
-    </fileSets>
-</assembly>
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (healchow) [inline inlong-distribution/src/main/assemblies/sort-connectors.xml:1]: Excuse, why change the format for this file?

**Comment by thesumery** [comment]
LGTM

File: inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/node/extract/DorisExtractNodeTest.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import org.apache.inlong.sort.formats.common.DecimalFormatInfo;
+import org.apache.inlong.sort.formats.common.DoubleFormatInfo;
+import org.apache.inlong.sort.formats.common.IntFormatInfo;
+import org.apache.inlong.sort.formats.common.StringFormatInfo;
+import org.apache.inlong.sort.protocol.FieldInfo;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class DorisExtractNodeTest extends SerializeBaseTest<DorisExtractNode> {
```
[the remark this answers is not in the document]
Comment (duowan1520) [reply inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/node/extract/DorisExtractNodeTest.java:33]: ok, i've dealt with it

File: inlong-sort/sort-connectors/doris/README.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
+# Overview
+The sort-connector-doris connector comes from doris-flink-connector officially provided by apache doris.
+The flink-doris-connector-1.13_2.12 version is currently introduced. If you need to use scala 2.11, you need to 
+download the flink-doris-connector source code of the apache doris community and create a new branch from 
+tag 1.13_2.11-1.0.3.
+
+# checkout and new branch
```
Comment (healchow) [inline inlong-sort/sort-connectors/doris/README.md:7]: Change to "Checkout a new branch".

File: inlong-sort/sort-connectors/doris/README.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+lzo-devel zlib-devel gcc gcc-c++`
+3. `tar zxvf thrift-0.13.0.tar.gz`
+4. `cd thrift-0.13.0`
+5. `./configure --without-tests`
+6. `make`
+7. `make install`
+
+Check the version after installation is complete：thrift --version
+
+# package flink-doris-connector
```
Comment (healchow) [inline inlong-sort/sort-connectors/doris/README.md:41]: Change to "Package flink-doris-connector".

File: inlong-sort/sort-connectors/doris/README.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+`mvn clean package -Dscala.version=2.11 -Dflink.version=1.13.5 -Dflink.minor.version=1.13
```
Comment (healchow) [inline inlong-sort/sort-connectors/doris/README.md:42]: For more than one line code, suggested use three \`, such as:
```
\``` (remove the \)
mvn clean package -Dscala.version=2.11 -Dflink.version=1.13.5 -Dflink.minor.version=1.13 
-Denv.THRIFT_BIN=/path/to/thrift
\```
```

File: inlong-sort/sort-core/src/test/java/org/apache/inlong/sort/parser/DorisExtractToMySqlLoadTest.java [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+                new FieldInfo("age", new IntFormatInfo()),
+                new FieldInfo("price", new DecimalFormatInfo()),
+                new FieldInfo("sale", new DoubleFormatInfo())
+        );
+        return new DorisExtractNode("1", "doris_input", fields,
+                null, null, "localhost:8030", "root",
+                "000000", "test.test1");
+    }
+
+    private Node buildMysqlLoadNode() {
```
Comment (healchow) [inline inlong-sort/sort-core/src/test/java/org/apache/inlong/sort/parser/DorisExtractToMySqlLoadTest.java:67]: `buildMySqlLoadNode`.

**Review by healchow — CHANGES_REQUESTED** [review]

**Review by EMsnap — APPROVED** [review]

**Comment by dockerzhang** [comment]
@duowan1520 
please fix the conflicting files and add LICENSE for the new dependency.

**Review by thesumery — APPROVED** [review]

**Comment by duowan1520** [comment]
> @duowan1520 please fix the conflicting files and add LICENSE for the new dependency.

ok. i have fixed it

**Review by dockerzhang — APPROVED** [review]

**Review by healchow — APPROVED** [review]
LGTM

**Pull request merged**